python/Ganga/GPIDev/Lib/File/MassStorageFile.py

```diff
@@ -163,28 +163,53 @@ def location(self):
             tmpLocations = self.locations
         return tmpLocations
 
-    def get(self):
+    def get(self, localPath=''):
         """
         Retrieves locally all files matching this MassStorageFile object pattern
         """
+        self._actual_copy()
 
-        to_location = self.localDir
-
-        if not os.path.isdir(self.localDir):
-            if self._getParent() is not None:
-                to_location = self.getJobObject().outputdir
-            else:
-                logger.error("%s is not a valid directory.... Please set the localDir attribute" % self.localDir)
-                return
+    def _actual_copy(self, localPath=''):
```

Comment:
I don't want to see  functions called "_actual_anything" as it suggests a poorly though-out separation of concerns. Having something like a set of functions called `copy`, `actual_copy` and `really_copy` which simply call each other in turn doesn't help anyone understand the code and obfuscates the purpose of any function.